opencog/nlp/scm/processing-utils.scm

```diff
@@ -127,3 +133,160 @@
 
 ; -----------------------------------------------------------------------
 ; -----------------------------------------------------------------------
+
+; -----------------------------------------------------------------------
+; A copy of relex-parse funtion modified for relex-to-logic(r2l) purposes.
+;
+; This version will assume R2L atoms are emitted by the RelEx server, and
+; wrap the R2L atoms inside a ReferenceLink with an InterpretationNode.
+;
+(define (r2l plain-txt)
+	; ---------------------------------------------------------------------
+	; Patterns used for spliting the string passed from relex server
+	(define pattern1 "; ##### END OF A PARSE #####")
+	(define pattern2 "; ##### START OF R2L #####")
+
+	; ---------------------------------------------------------------------
+	; Splits a string into substring delimited by the a pattern and returns a list
+	; of the substrings.
+	; TODO: make tail recursive, for efficency
+	(define (split-string a-pattern a-string)
+		(let ((a-match (string-match a-pattern a-string)))
+			(if (not a-match)
+				(list a-string)
+				(append (list (match:prefix a-match))
+					(split-string a-pattern (match:suffix a-match)))
+			)
+		)
+	)
+
+	; ---------------------------------------------------------------------
+	; returns the name of a ParseNode if the string has a ParseNode entry.
+	; returns "sentence@dbce9f0d-8b8a-4ea7-a8c8-d69abf05f810_parse_1" from the string
+	; "84590fb2e46c (ParseNode \"sentence@dbce9f0d-8b8a-4ea7-a8c8-d69abf05f810_parse_1\""
+	(define (parse-str a-string)
+		(substring (match:substring
+			(string-match "ParseNode \"sentence@[[:alnum:]_-]*" a-string)) 11)
+	)
```

Comment:
Well.. these codes are not really working on the atomspace, but is working on the string emitted by the RelEx server.

I believe something like this was discussed already when @AmeBel first pushed these codes at https://github.com/opencog/opencog/pull/920

On the other hand, unrelated to these codes, atom's name string comparison was always part of RelEx2Logic's incomplete design.  It is certainly necessary to fix, but needs well structured steps & goals.  R2L is still quite chaotic.